examples/3.x_api/pytorch/nlp/huggingface_models/language-modeling/quantization/static_quant/pt2e/run_clm_no_trainer.py

```diff
@@ -0,0 +1,153 @@
+                    help="Select from ['dynamic', 'static', 'weight-only']")
+parser.add_argument("--int8", action="store_true")
+parser.add_argument("--accuracy", action="store_true")
+parser.add_argument("--performance", action="store_true")
+parser.add_argument("--iters", default=100, type=int,
+                    help="For accuracy measurement only.")
+parser.add_argument("--batch_size", default=1, type=int,
+                    help="For accuracy measurement only.")
+parser.add_argument("--tasks", default="lambada_openai,hellaswag,winogrande,piqa,wikitext",
+                    type=str, help="tasks for accuracy validation")
+parser.add_argument("--peft_model_id", type=str, default=None, help="model_name_or_path of peft model")
+# =======================================
+
+args = parser.parse_args()
+
+
+def get_user_model():
+    torchscript = False
+    user_model = AutoModelForCausalLM.from_pretrained(
+        args.model,
+        torchscript=torchscript,  # torchscript will force `return_dict=False` to avoid jit errors
+        trust_remote_code=args.trust_remote_code,
+        revision=args.revision,
+    )
+    tokenizer = AutoTokenizer.from_pretrained(args.model)
+
+    if args.peft_model_id is not None:
+        from peft import PeftModel
+        user_model = PeftModel.from_pretrained(user_model, args.peft_model_id)
+
+    # to channels last
+    user_model = user_model.to(memory_format=torch.channels_last)
+    user_model.eval()
+    return user_model, tokenizer
+
+user_model, tokenizer = get_user_model()
+if args.quantize:
+    
+    from neural_compressor.torch.quantization import (
+            convert,
+            get_default_static_config,
+            prepare,
+        )
+    from neural_compressor.torch.export import export
+    from torch.export import Dim
+    # set TOKENIZERS_PARALLELISM to false
+    def get_example_inputs(tokenizer):
+        text = "Hello, welcome to LLM world."
+        encoded_input = tokenizer(text, return_tensors="pt")
+
+        example_inputs = encoded_input
+        # print(f"example_inputs: {example_inputs}")
+        input_ids = example_inputs["input_ids"]
+        input_ids_batch = torch.cat((input_ids, input_ids), dim=0)
+        print(f"input_ids_batch shape: {input_ids_batch.shape}")
+        tuple_inputs = (input_ids_batch,)
+        return tuple_inputs
+    # os.environ["TOKENIZERS_PARALLELISM"] = "false"
+    # torch._dynamo.config.cache_size_limit = 4 # set limitation if out of memory
+    batch = Dim(name="batch_size")
+    seq_len = Dim(name="seq_len")
+    dynamic_shapes = {"input_ids": (batch, seq_len)}
+    example_inputs = get_example_inputs(tokenizer)
+    exported_model = export(user_model, example_inputs=example_inputs, dynamic_shapes=dynamic_shapes)
+
+    quant_config = get_default_static_config()
+    # prepare
+    prepare_model = prepare(exported_model, quant_config)
+
+    # calibrate
+    for i in range(2):
```

Comment:
Is 2 batches enough?

Reply:
Verified that the accuracy remains unchanged as the batch size increases.

Reply:
How about exposing an arg for use to select the number of iterations?